src/renderer/kube-object.store.ts

```diff
@@ -279,7 +280,8 @@ export abstract class KubeObjectStore<T extends KubeObject> extends ItemStore<T>
   }
 
   async update(item: T, data: Partial<T>): Promise<T> {
-    const newItem = await item.update(data);
+    const rawItem = await item.update(data);
+    const newItem = new this.api.objectConstructor(rawItem);
```

Comment:
this behaviour seems a little different than it was in resource-applier.api, is this ok?